### PR TITLE
Fixed Universal/123Movie Provider

### PR DESCRIPTION
**Pull request by Teekzilla**
- Updated the search endpoint so that queries actually produce results.
- Stripped the years from TV show titles to find any results.

### CODE REVIEW

File: src/scrapers/providers/universal/123movie.js [hunk 1]
```diff
@@ -11,6 +11,9 @@ async function _123movie(req, sse){
     let season = req.query.season;
     let episode = req.query.episode;
 
+    // If it's a TV show, we need to strip the year
+    if(req.query.type === 'tv') queryTitle = queryTitle.substring(0, queryTitle.length - 7);
```
Comment (FlyPigsFly) [inline src/scrapers/providers/universal/123movie.js]: Would this be a better approach to removing the year?
```suggestion
    if(req.query.type === 'tv') queryTitle = queryTitle.replace(/\s\([0-9]{4}\)$/, "");
```

**Comment by SpajicM** [comment]
@FlyPigsFly on the other PR, @j0j00 described Regex as 'pretty fragile' (and imo, less readable) https://github.com/ApolloTVofficial/Claws/pull/19#discussion_r257851336

Also, according to https://jsperf.com/replace-or-substr - substring is faster.

**Comment by FlyPigsFly** [comment]
@SpajicM I don't believe @j0j00 we referring to RegExs in general as being fragile. This is a pretty basic regex and as for readable a comment describing what regex is doing could suffice. I choose a regex and replace because we are removing exactly what we want to remove versus blindly removing what we think should always be there.

**Comment by j0j00** [comment]
@FlyPigsFly is right.

@SpajicM, I think the Regex in this case is fine as it's simple and will work if the year exists in the string or not. That other one was because it was matching against JavaScript code which can be pretty dynamic, so I opted for evaluating it so the chances of it breaking was lower.

The ultimate solution would be for the client to send the year and title as separate query parameters and leave it up to the providers to use them as they deem fit.

**Comment by SpajicM** [comment]
Alright then. Ready for merge.
Meanwhile, I'll raise GH issues in Kamino repo for both type="anime" and this, so those (hopefully) temporary fixes don't stay in here, as sending the year but then removing it like this is pretty bad indeed.

**Comment by ddivad** [comment]
Sounds good :) I agree that having to remove isn't ideal in this case, but the majority of providers use year, so its not too bad. Separating them out would definitely be a more `robust` solution though.

**Pull request merged**